src/xdsl/irdl.py

```diff
@@ -372,7 +380,11 @@ def irdl_op_verify(op: Operation, operands: List[Tuple[str, OperandDef]],
                 operand_def.constr.verify(op.operands[current_operand].typ)
                 current_operand += 1
         else:
-            operand_def.constr.verify(op.operands[current_operand].typ)
+            try:
+                operand_def.constr.verify(op.operands[current_operand].typ)
+            except Exception as e:
+                error(op, f"{e}")
```

Comment:
Could we add maybe in the error which operand triggered the error?

Reply:
Done.